include/seqan3/range/container/small_string.hpp

```diff
@@ -309,6 +335,39 @@ class small_string : public small_vector<char, capacity_ + 1>
         return str();
     }
     //!\}
+
+    /*!\name Input/output
+     * \{
+     */
+
+    /*!\brief Formatted output for the seqan3::small_string.
+     * \param[in,out] os  The std::basic_ostream to write to.
+     * \param[in]     str The seqan3::small_string to read from.
+     * \returns `os`.
+     */
+    friend std::ostream & operator<<(std::ostream & os, small_string const & str)
+    {
+        os << str.str();
+        return os;
+    }
+
+    /*!\brief Formatted input for the seqan3::small_string.
+     * \param[in,out] is  The std::basic_istream to read from.
+     * \param[out]    str The seqan3::small_string to write to.
+     * \returns `is`.
+     */
+    friend std::istream & operator>>(std::istream & is, small_string & str)
+    {
+        str.erase();
+        std::string tmp;
+        is >> tmp;
+
+        auto min_size = std::min<typename std::string::size_type>(tmp.size(), str.max_size());
+        std::ranges::move(tmp.begin(), tmp.begin() + min_size, std::back_inserter(str));
```

Comment:
hm is this the expected behaviour, that the input is silently truncated?

Reply:
man 🤦‍♂ I was first adding the complete ugly implementation then thought tonight, ah just put it into a string and copy from there. But you are right, this might consume to much from the input.  So I need to get the ugly version back.

Reply:
we should assert though that the capacity is exceeded? 
I think trying to assign a string from stream that is larger than the capacity is UB so I don't know if this needs to be ugly?

Reply:
No, the stream has special functionality, for example only reading until a whitespace occurs or reading at most is.width  characters from the stream if width is set, etc. See here the details for the function https://en.cppreference.com/w/cpp/string/basic_string/operator_ltltgtgt. But if I parse it first into a string I cannot control the capacity there, so even if we only read as much characters as the capacity of small_string the basic string before might have consumed already more input. That's why I think that there isn't a more elegant strategy.